code/datums/elements/firestacker.dm

```diff
@@ -0,0 +1,40 @@
+/**
+  * Can be applied to /atom/movable subtypes to make them apply fire stacks to things they hit
+  */
+/datum/element/firestacker
+	element_flags = ELEMENT_DETACH
+	/// A list in format {atom/movable/owner, number}
+	/// Used to keep track of movables which want to apply a different number of fire stacks than default
+	var/list/amount_by_owner = list()
+
+/datum/element/firestacker/Attach(datum/target, amount)
+	. = ..()
+	if(!ismovableatom(target))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_MOVABLE_IMPACT, .proc/impact)
+	if(isitem(target))
+		RegisterSignal(target, COMSIG_ITEM_ATTACK, .proc/item_attack)
+		RegisterSignal(target, COMSIG_ITEM_ATTACK_SELF, .proc/item_attack_self)
+
+	if(amount) // If amount is not given we default to 1 and don't need to save it here
+		amount_by_owner[target] = amount
+
+/datum/element/firestacker/Detach(datum/source, force)
+	. = ..()
+	UnregisterSignal(source, list(COMSIG_MOVABLE_IMPACT, COMSIG_ITEM_ATTACK, COMSIG_ITEM_ATTACK_SELF))
+	amount_by_owner -= source
+
+/datum/element/firestacker/proc/stack_on(datum/owner, mob/living/target)
+	target.adjust_fire_stacks(amount_by_owner[owner] || 1)
```

Comment:
im pretty sure this will runtime if the owner isn't in the list

Reply:
It won't